src/crimson/osd/shard_services.cc

```diff
@@ -95,14 +96,126 @@ seastar::future<> PerShardState::broadcast_map_to_pgs(
   return seastar::parallel_for_each(
     pgs.begin(), pgs.end(),
     [=, &shard_services](auto& pg) {
+      auto old_map = pg.second->get_osdmap();
       return shard_services.start_operation<PGAdvanceMap>(
 	pg.second,
 	shard_services,
 	epoch,
-	PeeringCtx{}, false).second;
+	PeeringCtx{}, false, false).second.then(
```

Comment:
Suppose 1.0 splits into 1.0 and 1.1 at epoch 20.  It looks to me like this implementation advances pg 1.0 to 20, and then asynchronously splits it.  What prevents it from sending or processing peering messages in the meantime?

I suspect that this actually must be part of PGAdvanceMap.  The splits need to happen after the call to handle_advance_map for the splitting map and before the final call to handle_activate_map for the end of the sequence.

Reply:
That makes sense, I'll make the change.